### PR TITLE
migrate javaharness to pipes-shell-2

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java [hunk 1]
```diff
@@ -20,68 +21,60 @@
...
+        PortableJson content = jsonParser.parse(json);
+        String message = content.getString(FIELD_MESSAGE);
+        switch (message) {
+            case MESSAGE_READY:
+                window.console.log("Received 'ready' message");
+                break;
+            case MESSAGE_DATA:
+                String transactionId = String.valueOf(content.getInt(FIELD_TRANSACTION_ID));
+                if (inProgress.containsKey(transactionId)) {
+                    inProgress.get(transactionId).onData(transactionId, jsonParser.stringify(content.getObject(FIELD_DATA)));
```
Comment (cromwellian) [inline src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java]: I wonder if we need a null check for content.getObject() here?

File: particles/PipeApps2/PipeApps.recipes [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import 'MapsAutofill.recipes'
 import 'PeopleAutofill.recipes'
-import 'ArtistAutofill.recipes'
-import 'Caption.recipes'
-import 'UpdateMe.recipes'
-import 'ArtistPipe.recipes'
+// import 'ArtistAutofill.recipes'
```
Comment (shaper) [inline particles/PipeApps2/PipeApps.recipes]: Is this intentional?

File: particles/PipeApps2/canonical.recipes [hunk 3]
```diff
@@ -0,0 +1 @@
+import 'PipeApps.recipes'
```
Comment (shaper) [inline particles/PipeApps2/canonical.recipes]: Same -- intentional? Why is `canonical.recipes` blank currently? @yuangu-google

File: src/javaharness/java/arcs/api/ShellApi.java [hunk 4]
```diff
@@ -1,11 +1,5 @@
...
 
 public interface ShellApi {
-    void observeEntity(String entityJson);
-
-    String receiveEntity(String entityJson);
-
-    void chooseSuggestion(String suggestion);
-
-    void postMessage(String msgToSendToHost);
+    String receive(String json);
```
Comment (shaper) [inline src/javaharness/java/arcs/api/ShellApi.java]: Can we add brief interface doc comment, and comment on this method? I can never remember when reading this which side of the pipe this is. Who's receiving what? Is this Android Java receiving a message from the Shell, or the Shell receiving a message from Android Java?

File: src/javaharness/java/arcs/webimpl/WebHarnessComponent.java [hunk 5]
```diff
@@ -3,7 +3,9 @@
 import arcs.api.ArcsEnvironment;
 import arcs.nativeparticles.NativeParticlesModule;
 import dagger.Component;
+import javax.inject.Singleton;
 
+@Singleton
```
Comment (shaper) [inline src/javaharness/java/arcs/webimpl/WebHarnessComponent.java]: Why does this have to be a `Singleton` now, but wasn't before?

File: src/javaharness/java/arcs/webimpl/WebHarnessModule.java [hunk 6]
```diff
@@ -7,15 +7,17 @@
...
 import java.util.Map;
+import javax.inject.Singleton;
 
 @Module
 public abstract class WebHarnessModule {
 
   @Binds
   public abstract ArcsEnvironment provideStandaloneWebArcsEnvironment(ShellApiBasedArcsEnvironment impl);
 
+  @Singleton
```
Comment (shaper) [inline src/javaharness/java/arcs/webimpl/WebHarnessModule.java]: Similar to above comment -- why does it need to be `Singleton` now? Just want to understand backstory.

File: src/javaharness/java/arcs/webimpl/WebHarnessComponent.java (hunk 5 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/webimpl/WebHarnessComponent.java]: this change was actually already merged in https://github.com/PolymerLabs/arcs/pull/3295 - without it the `public static Map<String, ArcsEnvironment.DataListener> provideInProgressListeners()` always provides a new map and the listeners are not registered - this was a preexisting problem in the javaharness, and is unrelated to the migration.

File: particles/PipeApps2/canonical.recipes (hunk 3 above)
Comment (mmandlis) [reply particles/PipeApps2/canonical.recipes]: uncommented (and got rid of canonical.recipes - Scott says it is redundant)

File: src/javaharness/java/arcs/api/ShellApi.java (hunk 4 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/api/ShellApi.java]: added comment here and in DeviceClient.java

File: particles/PipeApps2/PipeApps.recipes (hunk 2 above)
Comment (mmandlis) [reply particles/PipeApps2/PipeApps.recipes]: thanks, uncommented (and got rid of canonical.recipes - Scott says it is redundant)

File: src/javaharness/java/arcs/webimpl/WebHarnessModule.java (hunk 6 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/webimpl/WebHarnessModule.java]: (this is part of the same change, replied above)

File: src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java (hunk 1 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java]: done

**Comment by mmandlis** [comment]
Thanks, addressed comments, @shaper PTAL!

**Pull request merged**